cellengine/resources/gate.py

```diff
@@ -1,158 +1,74 @@
-            create_population (bool): Automatically create corresponding population.
-        """
-        if type(gates) is list:
-            return [cls._build_gate(gate) for gate in gates]
+    def update(self):
+        """Save changes to this Gate to CellEngine.
+        If this gate does not exist, it will be created."""
+        if not self._id or not is_valid_id(self._id):
+            gate = self.client.create(self)
         else:
-            return cls._build_gate(gates)
-
-    @classmethod
-    def _build_gate(cls, gate):
-        """Get the gate type and return instance of the correct subclass."""
-        module = importlib.import_module(__name__)
-        gate, _ = cls._separate_gate_and_population(gate)
-        gate_type = getattr(module, gate["type"])
-        return gate_type(properties=gate)
+            gate = self.client.update(self)
+        self.__setstate__(gate.__getstate__())  # type: ignore
 
     @classmethod
-    def _separate_gate_and_population(cls, gate):
-        try:
-            if "gate" in gate.keys():
-                return gate["gate"], [k for k in gate.keys() if k != "gate"]
-            else:
-                return gate, None
-        except KeyError:
-            raise ValueError("Gate payload format is invalid")
-
-    @staticmethod
-    def delete_gates(
-        experiment_id, _id: str = None, gid: str = None, exclude: str = None
-    ) -> None:
-        """Deletes a gate or a tailored gate family.
-
-        Specify the top-level gid when working with compound gates (specifying
-        the gid of a sector (i.e. one listed in ``model.gids``) will result in
-        no gates being deleted). If ``_id`` is specified, only that gate will be
-        deleted, regardless of the other parameters specified. May be called as
-        a static method on `cellengine.Gate` or on an `Experiment` instance.
-
-        Args:
-            experiment_id (str): ID of experiment.
-            _id (str): ID of gate.
-            gid (str): ID of gate family.
-            exclude (str): Gate ID to exclude from deletion.
-
-        Example:
-            ```python
-            cellengine.Gate.delete_gates(experiment_id, gid = <gate family ID>)
-            # or
-            experiment.delete_gates(_id = <gate ID>)
-            # or
-            experiment.delete_gates(gid = <gate family ID>, exclude = <gate ID>)
-            ```
+    def factory(cls, gate):
+        """Create a gate from a dict of keys."""
+        if "_id" not in gate.keys():
+            gate["_id"] = None
+        return converter.structure(gate, Gate)
 
-        Returns:
-            None
-
-        """
-        ce.APIClient().delete_gate(experiment_id, _id, gid, exclude)
+    def delete(self):
+        self.client.delete_gate(self.experiment_id, self._id)
 
     @staticmethod
-    def update_gate_family(experiment_id, gid: str, body: Dict):
+    def update_family(experiment_id, gid: str, body: Dict):
```

Comment:
Let's keep the original name to avoid a breaking change. Can add an alias if you want.